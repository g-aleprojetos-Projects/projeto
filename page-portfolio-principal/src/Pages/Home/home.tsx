import React from 'react';
import manutencao from 'assets/image/manutencao.svg';

import * as S from './home.styles';

export const Home = () => {
  return (
    <S.Container data-testid={'teste_home'}>
      <S.Imagem data-testid={'teste_imagem'} src={manutencao} />
    </S.Container>
  );
};
