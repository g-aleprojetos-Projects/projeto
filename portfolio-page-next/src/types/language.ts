export enum LanguageEnum {
  en_US = 'en-US',
  fr_FR = 'fr-FR',
  pt_BR = 'pt-BR',
}
