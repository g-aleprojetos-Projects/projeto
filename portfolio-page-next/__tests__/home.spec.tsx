import React from 'react';
import {customRender, screen} from './test-setup';

import Home from '@/app/[locale]/page';
import '@testing-library/jest-dom';
import {LanguageEnum} from '@/types';

describe('Home', () => {
  describe('Renderização', () => {
    test('Home DEVE ser igual ao snapshot', () => {
      const {container} = customRender(<Home />);
      expect(container).toMatchSnapshot();
    });

    test('DEVE renderizar a pagina "Home"', () => {
      customRender(<Home />);

      expect(screen.getByTestId('teste_home')).toBeInTheDocument();
    });

    test('DEVE renderizar o texto em Inglês "Hello world!"', () => {
      customRender(<Home />);

      expect(screen.getByText(/Hello world!/i)).toBeInTheDocument();
    });

    test('DEVE renderizar o texto em Francês "Bonjour le Monde!"', () => {
      customRender(<Home />, LanguageEnum.fr_FR);

      expect(screen.getByText(/Bonjour le Monde!/i)).toBeInTheDocument();
    });

    test('DEVE renderizar o texto em Português "Olá Mundo!"', () => {
      customRender(<Home />, LanguageEnum.pt_BR);

      expect(screen.getByText(/Olá Mundo!/i)).toBeInTheDocument();
    });
  });
});
