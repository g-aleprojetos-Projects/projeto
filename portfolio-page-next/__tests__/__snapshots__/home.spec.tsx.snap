// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Renderização Home DEVE ser igual ao snapshot 1`] = `
<div>
  <h1
    data-testid="teste_home"
  >
    Hello world!
  </h1>
</div>
`;
